File: sea-shell-service-immutable/src/main/java/org/jesperancinha/shell/webflux/immutable/data/SeaShellDto.java
package org.jesperancinha.shell.webflux.immutable.data;


import lombok.Builder;

import java.util.List;

public record SeaShellDto(
        String name,
        String scientificName,
        String slogan,
        List<SeaShellPersonDto> persons,
        List<SeaShellCostumeDto> costumes,
        List<Long> personIds,
        List<Long> costumeIds) {

    @Builder
    public SeaShellDto {
    }
}

File: sea-shell-service-immutable/src/main/kotlin/org/jesperancinha/shell/webflux/immutable/data/SeaShellDto.kt
package org.jesperancinha.shell.webflux.immutable.data

import lombok.Builder

@JvmRecord
data class SeaShellDto @Builder constructor(
    val name: String,
    val scientificName: String,
    val slogan: String,
    val persons: List<SeaShellPersonDto>,
    val costumes: List<SeaShellCostumeDto>,
    val personIds: List<Long>,
    val costumeIds: List<Long>
)